darts/dataprocessing/dtw/dtw.py

```diff
@@ -0,0 +1,338 @@
+        """
+        Warps the two time series according to the warp path returned by .path(), which minimizes the pair-wise distance.
+        This will bring two time series that are out-of-phase back into phase.
+
+        Returns
+        -------
+        (TimeSeries, TimeSeries)
+            Two new TimeSeries instances of the same length, indexed by pd.RangeIndex.
+        """
+
+        series1 = self.series1
+        series2 = self.series2
+
+        xa1 = series1.data_array(copy=False)
+        xa2 = series2.data_array(copy=False)
+
+        path = self.path()
+
+        warped_series1 = xa1[path[:, 0]]
+        warped_series2 = xa2[path[:, 1]]
+
+        time_dim1 = series1._time_dim
+        time_dim2 = series2._time_dim
+
+        range_index = True
+
+        if range_index:
+            warped_series1 = warped_series1.reset_index(dims_or_levels=time_dim1)
+            warped_series2 = warped_series2.reset_index(dims_or_levels=time_dim2)
+
+        # todo: prevent time information being lost after warping
+        # Applying time index from series1 to series2 (take_dates = True) is disabled for consistency reasons
+        # Applying the warp path to the dates directly will result in duplicate dates
+        # and hence values being lost when converting back to a TimeSeries.
+        # As a result series1 will not be warped, whereas series2 will be.
+        # It could also cause the two series to have different lengths, if len(series1) < len(series2)
+        # One could generate intermediate dates, but then the series would not have a consistent frequency
+
+        take_dates = False
+        if take_dates:
+            time_index = warped_series1[time_dim1]
+            time_index = time_index.rename({time_dim1: time_dim2})
+            warped_series2[time_dim2] = time_index
+
+        return TimeSeries.from_xarray(warped_series1), TimeSeries.from_xarray(warped_series2)
+
+
+def default_distance_multi(x_values: np.ndarray, y_values: np.ndarray):
+    return np.sum(np.abs(x_values - y_values))
+
+
+def default_distance_uni(x_value: float, y_value: float):
+    return abs(x_value - y_value)
+
+
+def dtw(series1: TimeSeries,
+        series2: TimeSeries,
+        window: Window = NoWindow(),
+        distance: Union[DistanceFunc, None] = None,
+        multi_grid_radius: int = -1
+        ) -> DTWAlignment:
+    """
+    Determines the optimal alignment between two time series series1 and series2, according to the Dynamic Time Warping algorithm.
+    The alignment minimizes the distance between pair-wise elements after warping. All elements in the two series are matched
+    and are in strictly monotonically increasing order. Considers only the values in the series, ignoring the time axis.
+
+    Dynamic Time Warping can be applied to determine how closely two time series correspond,
+    irrespective of phase, length or speed differences.
+
+    Parameters
+    ----------
+    series1
+        `TimeSeries`
```

Comment:
Could you put a bit more text here and also for series2. Just so that we have a short description for every parameter.